### PR TITLE
Add emulating authorizedRoles in XB #7247

### PR DESCRIPTION
**Pull request by alhambrav**
### Ticket reference or full description of what's in the PR
Add emulating authorizedRoles in XB https://github.com/craftercms/craftercms/issues/7247

### CODE REVIEW

**Review by sumerjabri — APPROVED** [review]

File: source/by-role/developer/common/targeting.rst [hunk 1]
```diff
@@ -639,3 +641,134 @@ in the ``Entertainment`` category page.
...
+                      <label>Anonymous</label>
+                      <value>anonymous</value>
+                    </value>
+                    <value>
+                      <label>Role A</label>
+                      <value>RoleA</value>
+                    </value>
+                    <value>
+                      <label>Role B</label>
+                      <value>RoleB</value>
```
Comment (rart) [inline source/by-role/developer/common/targeting.rst:713]: It may not be part of this PR but one detail is that I don't see any mention to where these roles are defined. For example, is "anonymous" something understood by the system or is that defined somewhere? Or is it simply that it doesn't match one of the defined roles so it is essentially interpreted as anonymous?

Similarly for Role(A|B), are those defined somewhere?

**Review by rart — APPROVED** [review]
